imap_processing/lo/l0/data_classes/science_counts.py

```diff
@@ -0,0 +1,285 @@
+        Triple coincidence rates for TOF0 and TOF1. 2D array Azimuth(60), Energy(7)
+    TOF0_TOF2: numpy.ndarray
+        Triple coincidence rates for TOF0 and TOF2. 2D array Azimuth(60), Energy(7)
+    TOF1_TOF2: numpy.ndarray
+        Triple coincidence rates for TOF1 and TOF2. 2D array Azimuth(60), Energy(7)
+    SILVER: numpy.ndarray
+        Triple coincidence rates for TOF0, TOF1, TOF2 and TOF3.
+        2D array Azimuth(60), Energy(7)
+    DISC_TOF0: numpy.ndarray
+        Discarded rates for TOF0 value less than TOF0 threshold setting.
+        2D array Azimuth(6), Energy(7)
+    DISC_TOF1: numpy.ndarray
+        Discarded rates for TOF1 value less than TOF1 threshold setting.
+        2D array Azimuth(6), Energy(7)
+    DISC_TOF2: numpy.ndarray
+        Discarded rates for TOF2 value less than TOF2 threshold setting.
+        2D array Azimuth(6), Energy(7)
+    DISC_TOF3: numpy.ndarray
+        Discarded rates for TOF3 value less than TOF3 threshold setting.
+        2D array Azimuth(6), Energy(7)
+    POS0: numpy.ndarray
+        Postition rate counts for Ion anode B0. 2D array Azimuth(6), Energy(7)
+    POS1: numpy.ndarray
+        Position rate counts for Ion anode B1. 2D array Azimuth(6), Energy(7)
+    POS2: numpy.ndarray
+        Position rate counts for Ion anode B2. 2D array Azimuth(6), Energy(7)
+    POS3: numpy.ndarray
+        Position rate counts for Ion anode B3. 2D array Azimuth(6), Energy(7)
+    HYDROGEN: numpy.ndarray
+        Hydrogen species histogram. 2D array Azimuth(60), Energy(7)
+    OXYGEN: numpy.ndarray
+        Oxygen species histogram. 2D array Azimuth(60), Energy(7)
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
+        Uses the CCSDS packet, version of the software, and
+        the name of the packet file to parse and store data for
+        the Science Count packet.
+    """
+
+    SHCOARSE: int
+    SCI_CNT: str
+    START_A: np.ndarray
+    START_C: np.ndarray
+    STOP_B0: np.ndarray
+    STOP_B3: np.ndarray
+    TOF0: np.ndarray
+    TOF1: np.ndarray
+    TOF2: np.ndarray
+    TOF3: np.ndarray
+    TOF0_TOF1: np.ndarray
+    TOF0_TOF2: np.ndarray
+    TOF1_TOF2: np.ndarray
+    SILVER: np.ndarray
+    DISC_TOF0: np.ndarray
+    DISC_TOF1: np.ndarray
+    DISC_TOF2: np.ndarray
+    DISC_TOF3: np.ndarray
+    POS0: np.ndarray
+    POS1: np.ndarray
+    POS2: np.ndarray
+    POS3: np.ndarray
+    HYDROGEN: np.ndarray
+    OXYGEN: np.ndarray
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
+        self.parse_data(packet)
+        self._parse_binary()
+        pass
+
+    def _parse_binary(self):
+        """Parse the science count binary chunk for each section of data."""
+        # make a bit stream containing the binary for the entire
+        # chunk of Science Count packet data
+        bitstream = ConstBitStream(bin=self.SCI_CNT)
```

Comment:
Is `self.SCI_CNT` a Python string, or is it a bitstring initially? You might be able to just re-use the self.SCI_CNT directly here rather than creating a new stream?

Reply:
Thanks for clarifying! I was hoping that was the case and I think we might be able to do some other faster manipulations rather than going back to bitstring within these functions then. We can chat more later about this.